packages/yew/src/suspense/hooks.rs

```diff
@@ -60,6 +96,54 @@ mod feat_futures {
             Err((*suspension).clone())
         }
     }
+
+    /// Use the result of an async computation with dependencies, suspending while waiting.
+    ///
+    /// Awaits the future returned from `f` for the latest `deps`. Even if the future is immediately
+    /// [ready], the hook suspends at least once. If the dependencies
+    /// change while a future is still pending, the result is never used. This guarantees that your
+    /// component always sees up-to-date values while it is not suspended.
+    ///
+    /// [ready]: std::task::Poll::Ready
+    #[hook]
+    pub fn use_future_with_deps<F, D, T, O>(f: F, deps: D) -> SuspensionResult<UseFutureHandle<O>>
+    where
+        F: FnOnce(&D) -> T,
```

Comment:
I want to say that this is also a problem in `use_memo` and should be solved there as well.

Reply:
How's this a problem with `use_memo`? There's no future there that lives for `'static` and the init function is not `'static`. 

```rust
use_memo(|deps| {
    // use deps here
}, deps)
``` should be valid
but 

```rust
use_future_with_deps(|deps| async move {
    // invalid because &deps is used in 'static future
}, deps)
```

Reply:
Oh excuse me, I didn't mean it's the same problem, but only a similar one. If one wants to `use_memo`, e.g. for a `Timeout`, then one also has to `clone()` the `&D` given as argument, even though it already is `Rc<D>` behind the scenes and could be cloned more cheaply. But I guess it's more relevant in this case as moving into the future is almost the whole point :)

Reply:
I guess exposing the Rc in use_memo wouldn't hurt. We should expose it here though. Can you please make that change? The clone is needed to use this hook in any meaningful capacity